radix-engine/src/model/epoch_manager/executables.rs

```diff
@@ -416,3 +456,100 @@ impl EpochManager {
         ))]
     }
 }
+
+pub struct ValidatorRegisterExecutable(RENodeId);
+
+impl<W: WasmEngine> ExecutableInvocation<W> for ValidatorRegisterInvocation {
+    type Exec = ValidatorRegisterExecutable;
+
+    fn resolve<D: ResolverApi<W>>(
+        self,
+        deref: &mut D,
+    ) -> Result<(ResolvedActor, CallFrameUpdate, Self::Exec), RuntimeError>
+    where
+        Self: Sized,
+    {
+        let mut call_frame_update = CallFrameUpdate::empty();
+        let receiver = RENodeId::Global(GlobalAddress::System(self.receiver));
+        let resolved_receiver = deref_and_update(receiver, &mut call_frame_update, deref)?;
+
+        let actor = ResolvedActor::method(
+            NativeFn::Validator(ValidatorFn::Register),
+            resolved_receiver,
+        );
+        let executor = ValidatorRegisterExecutable(resolved_receiver.receiver);
+        Ok((actor, call_frame_update, executor))
+    }
+}
+
+impl Executor for ValidatorRegisterExecutable {
+    type Output = ();
+
+    fn execute<Y>(self, api: &mut Y) -> Result<((), CallFrameUpdate), RuntimeError>
+    where
+        Y: SystemApi + InvokableModel<RuntimeError>,
+    {
+        let offset = SubstateOffset::Validator(ValidatorOffset::Validator);
+        let handle = api.lock_substate(self.0, offset, LockFlags::read_only())?;
+        let substate = api.get_ref(handle)?;
+        let validator = substate.validator();
+        let invocation = EpochManagerUpdateValidatorInvocation {
+            receiver: validator.manager,
+            validator_address: validator.address,
+            key: validator.key,
+            register: true,
+        };
+
+        api.invoke(invocation)?;
```

Comment:
Answered my previous question.

I wonder if we really need these methods inside the engine though. Now, we need to configure auth in two places.

More like a client-side abstraction.